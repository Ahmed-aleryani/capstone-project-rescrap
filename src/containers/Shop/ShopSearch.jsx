import React from 'react';
import { useTranslation } from 'react-i18next';

const ShopSearch = ({ searchText,text }) => {
  const {t}=useTranslation();

 

  const onSubmit = (e) => {
    e.preventDefault();
    searchText(text);
  };

  return (
    <div className="max-w-sm rounded overflow-hidden my-10 mx-auto">
      <form onSubmit={onSubmit} className="w-full max-w-sm">
        <div className="flex items-center border-b border-b-2 border-green-500 py-2">
          <input
            onChange={(e) => searchText(e.target.value)}
            className="appearance-none bg-transparent border-none w-full text-gray-600 mr-3 py-1 px-2 leading-tight focus:outline-none"
            type="text"
            placeholder={ t('translation:pages.shop.shopsearch.placeholder') }
            value={text}
          />
          {/* <button
            className="flex-shrink-0 bg-green-500 hover:bg-green-600 border-green-500 hover:border-green-600 text-sm border-4 text-white py-1 px-2 rounded"
            type="submit"
          >
            { t('translation:pages.shop.shopsearch.search-btn') }
          </button> */}
        </div>
      </form>
    </div>
  );
};

export default ShopSearch;
