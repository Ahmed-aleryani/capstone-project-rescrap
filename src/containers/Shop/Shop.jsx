import React, { useState, useEffect } from 'react';
import Card from '../../components/Card/Card';
import Button from '../../components/Button/Button';
import Title from '../../components/Title/Title';
import ShopSearch from './ShopSearch';
import firebase from '../../firebase.config';


// for (let i = 0; i < 16; i++) {
//   cards.push(
//     <Card
//       cardImageSrc="https://source.unsplash.com/random"
//       cardImageAlt="Rescrap Product"
//       productDetailsLink="#"
//       cardTitle="Plastic Bottels"
//       typeOfProduct="Plastic"
//       cardLocation="Sana'a"
//       cardPrice="777YER"
//     />
//   );
// }

const Shop = () => {
  // const cards = [];
  const [cards,setCards]=useState([]);
  const [start, setStart] = useState(0);
  useEffect(() => {
    setStart(0);
  }, []);

  useEffect(() => {
    console.log(cards);
  }, [cards])

  useEffect(async () => {

   firebase.database().ref('posts/').on('value', async function(snapshot) {


    const fetchImage =async (imageID)=>{
      const ref =  firebase.storage().ref('/posts-images/'+imageID);
      const url = await ref.getDownloadURL();
      return url;
    }
    const posts = snapshot.val();
    const keys=Object.keys(snapshot.val());
    
    console.log(keys);
    for(let i =0;i<keys.length;i++){
      const post =posts[keys[i]];
      console.log();
      const img=await fetchImage(posts[keys[i]].productImageKey);
           setCards(
            (previousState) => {
              return [...previousState,<Card
                cardImageSrc={img}
                cardImageAlt={post.description}
                productDetailsLink={"/shop/"+keys[i]}
                cardTitle="Plastic Bottels"
                typeOfProduct="Plastic"
                cardLocation="Sana'a"
                cardPrice="777YER"
                   />];
          }
         );
    }
    // snapshot.forEach( (element,key)=>{
    //   // const post = element.toJSON();
    //   // console.log(post);
    //   // console.log(key)
 

    //   // setCards([...cards,(<Card
    //   //   cardImageSrc={url}
    //   //   cardImageAlt={post.description}
    //   //   productDetailsLink={"/shop/"}
    //   //   cardTitle="Plastic Bottels"
    //   //   typeOfProduct="Plastic"
    //   //   cardLocation="Sana'a"
    //   //   cardPrice="777YER"
    //   // />)]);
    
    // })

  
  }
   )
  }, [])

  return (
    <div className="container mx-auto">
      <Title
        classes="uppercase font-semibold leading-5  tracking-widest text-green-500 text-center p-2 my-2 "
        text="SHOP PRODUCTS"
      />
      <ShopSearch />

      <div className="mb-4">
        <div className=" flex flex-wrap justify-between">
          {
            cards
          }
        </div>
      </div>
      <Button
        btnClasses="container block capitaliz mx-auto shadow bg-green-500 hover:bg-green-600 focus:shadow-outline focus:outline-none text-white text-lg py-3 px-10 rounded my-2"
        btnText="See More"
        btnClickHandler=""
      />
    </div>
  );
};

export default Shop;
