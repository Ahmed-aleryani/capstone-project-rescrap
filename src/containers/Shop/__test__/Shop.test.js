import React from 'react';
import Shop from '../../Shop/Shop';
import renderer from 'react-test-renderer';

it('matches snapshot', () => {
  const tree = renderer.create(<Shop />).toJSON();
  expect(tree).toMatchSnapshot();
});
