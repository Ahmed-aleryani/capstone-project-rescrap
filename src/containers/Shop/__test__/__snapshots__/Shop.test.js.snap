// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container mx-auto"
>
  <h1
    className="text-3xl normaluppercase font-semibold leading-5  tracking-widest text-green-500 text-center p-2 my-2 "
  >
    translation:pages.shop.title
  </h1>
  <div
    className="max-w-sm rounded overflow-hidden my-10 mx-auto"
  >
    <form
      className="w-full max-w-sm"
      onSubmit={[Function]}
    >
      <div
        className="flex items-center border-b border-b-2 border-green-500 py-2"
      >
        <input
          className="appearance-none bg-transparent border-none w-full text-gray-600 mr-3 py-1 px-2 leading-tight focus:outline-none"
          onChange={[Function]}
          placeholder="translation:pages.shop.shopsearch.placeholder"
          type="text"
        />
        <button
          className="flex-shrink-0 bg-green-500 hover:bg-green-600 border-green-500 hover:border-green-600 text-sm border-4 text-white py-1 px-2 rounded"
          type="submit"
        >
          translation:pages.shop.shopsearch.search-btn
        </button>
      </div>
    </form>
  </div>
  <div
    className="mb-4"
  >
    <div
      className=" flex flex-wrap justify-between"
    />
  </div>
  <button
    className="rounded-full py-2 px-4container block capitaliz mx-auto shadow bg-green-500 hover:bg-green-600 focus:shadow-outline focus:outline-none text-white text-lg py-3 px-10 rounded my-2"
    onClick=""
  >
    translation:pages.shop.btn-see-more
  </button>
</div>
`;
